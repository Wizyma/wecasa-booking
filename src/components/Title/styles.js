import styled from 'styled-components';

export const StyledTitle = variant =>
  styled[variant]({
    width: '100%',
    fontFamily: 'Open Sans',
  });
