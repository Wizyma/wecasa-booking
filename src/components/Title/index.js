import { arrayOf, node, oneOfType, string } from 'prop-types';
import React from 'react';

import { StyledTitle } from './styles';

const Title = ({ children, variant }) => {
  const STitle = StyledTitle(variant);
  return <STitle>{children}</STitle>;
};

Title.propTypes = {
  children: oneOfType([node, arrayOf(node)]),
  variant: string,
};

Title.defaultProps = {
  children: null,
  variant: 'h1',
};

export default Title;
