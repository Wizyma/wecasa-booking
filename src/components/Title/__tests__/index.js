import React from 'react';
import { cleanup, render } from 'react-testing-library';

import Title from '../';

afterEach(cleanup);

describe('Title component', () => {
  it('Title should match snapshot', () => {
    const { baseElement } = render(
      <Title>
        <p>Hello World</p>
      </Title>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should return an empty div when no children', () => {
    const { baseElement } = render(<Title />);

    expect(baseElement).toMatchSnapshot();
  });
});
