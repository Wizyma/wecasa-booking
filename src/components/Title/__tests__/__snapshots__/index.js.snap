// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component Title should match snapshot 1`] = `
<body>
  <div>
    <h1
      class="sc-bdVaJa kbLwTv"
    >
      <p>
        Hello World
      </p>
    </h1>
  </div>
</body>
`;

exports[`Title component should return an empty div when no children 1`] = `
<body>
  <div />
</body>
`;
